# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/interfaces/security/nsIContentSecurityPolicy.idl
# Commit: e21ad27bfd0a
# Full Hash: e21ad27bfd0a2fef90919101eaef5aa5af1cc6c2
# Author: Christoph Kerschbaumer <ckerschb@christophkerschbaumer.com>
# Date: 2019-10-22 21:43:14
# Regressor Bug: 1584993
# File Overlap Count: 1
# Description:
#   Bug 1584993: Make CSP frame-ancestors work with fission enabled. r=jkt,farre,valentin
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D49147
# ==============================================================================

diff -r d9d678e7422e -r e21ad27bfd0a dom/interfaces/security/nsIContentSecurityPolicy.idl
--- a/dom/interfaces/security/nsIContentSecurityPolicy.idl	Tue Oct 22 09:37:38 2019 +0000
+++ b/dom/interfaces/security/nsIContentSecurityPolicy.idl	Tue Oct 22 10:57:43 2019 +0000
@@ -6,8 +6,8 @@
 #include "nsIContentPolicy.idl"
 
 interface nsIURI;
-interface nsIDocShell;
 interface nsIEventTarget;
+interface nsILoadInfo;
 interface nsIPrincipal;
 interface nsICSPEventListener;
 
@@ -275,14 +275,14 @@
    * NOTE: Calls to this may trigger violation reports when queried, so this
    * value should not be cached.
    *
-   * @param docShell
-   *    containing the protected resource
+   * @param aLoadInfo
+   *    The loadinfo of the channel containing the protected resource
    * @return
    *    true if the frame's ancestors are all allowed by policy (except for
    *    report-only policies, which will send reports and then return true
    *    here when violated).
    */
-  boolean permitsAncestry(in nsIDocShell docShell);
+  boolean permitsAncestry(in nsILoadInfo aLoadInfo);
 
 
   /**